quint/src/types/substitutions.ts

```diff
@@ -51,69 +52,79 @@ export function compose(table: LookupTable, s1: Substitutions, s2: Substitutions
  *          given type
  */
 export function applySubstitution(table: LookupTable, subs: Substitutions, t: QuintType): QuintType {
-  switch (t.kind) {
-    case 'var': {
-      const sub = subs.find(s => s.name === t.name)
-      if (sub && sub.kind === 'type') {
-        return sub.value
-      } else {
-        return t
+  // We cannot assign the result of a switch to a value, so we use an arrow function instead
+  const singleApplication = () => {
+    switch (t.kind) {
+      case 'var': {
```

Comment:
idk what happen with the diff, but inside the switch, the only change is indentation.